dbcon/joblist/tuple-bps.cpp

```diff
@@ -1512,7 +1512,7 @@ bool TupleBPS::processSingleFilterString(int8_t BOP, int8_t colWidth, T val, con
                 break;
 
             case 16:
-                bigVal2 = *((int128_t*) filterString);
+                datatypes::TSInt128::assignPtrPtr(&bigVal2, (int128_t*) filterString);
```

Comment:
I suggest to make bigVal2 datatypes::TSInt128 and replace assignPtrPtr here with bigVal2 = datatypes::TSInt128(reinterpret_cast<int128_t*>(filterString)). Then compareSingleValue() will have bigVal2.getValue() as the last argument at the line 1527.

Reply:
Thanks for this suggestion, I have updated the code.